os/arch/arm/src/amebasmart/amebasmart_usb.c

```diff
@@ -0,0 +1,803 @@
+		buf[6] = lc->data_type;
+		break;
+
+	case CDC_SET_CONTROL_LINE_STATE:
+		/*
+		wValue:	wValue, Control Signal Bitmap
+				D2-15:	Reserved, 0
+				D1:	RTS, 0 - Deactivate, 1 - Activate
+				D0:	DTR, 0 - Not Present, 1 - Present
+		*/
+		cdc_acm_ctrl_line_state = req->wValue;
+		if (cdc_acm_ctrl_line_state & 0x01) {
+			dbg("\n[CDC] VCOM port activated\n");
+#if CONFIG_AMEBASMART_CDC_ACM_NOTIFY
+			usbd_cdc_acm_notify_serial_state(CDC_ACM_CTRL_DSR | CDC_ACM_CTRL_DCD);
+#endif
+		}
+		break;
+
+	case CDC_SEND_BREAK:
+		/* Do nothing */
+		break;
+
+	default:
+		break;
+	}
+
+	return HAL_OK;
+}
+
+static void amebasmart_cdc_acm_cb_status_changed(uint8_t status)
+{
+	// struct amebasmart_usbdev_s *dev = (struct amebasmart_usbdev_s *)priv;
+	// DEBUGASSERT(dev);
+
+#if CONFIG_AMEBASMART_USBD_CDC_ACM_HOTPLUG
+	cdc_acm_attach_status = status;
+	rtw_up_sema(&cdc_acm_attach_status_changed_sema);
+#endif
+}
+
+
+static int amebasmart_usbd_cdc_acm_init(struct amebasmart_usbdev_s *priv, usbd_cdc_acm_cb_t *cb)
+{
+	struct amebasmart_usbdev_s *dev = (struct amebasmart_usbdev_s *)priv;
+	int ret = -1;
+
+	ret = usbd_cdc_acm_init(CONFIG_CDC_ACM_BULK_OUT_XFER_SIZE, CONFIG_CDC_ACM_BULK_IN_XFER_SIZE, &amebasmart_cdc_acm_cb);
+	if (ret != 0) {
+		dbg("USB ACM init failed\n");
+		amebasmart_usbd_cdc_acm_deinit(dev);
+	}
+	return ret;
+}
+
+static amebasmart_usbd_cdc_acm_deinit(struct amebasmart_usbdev_s *priv)
+{
+	struct amebasmart_usbdev_s *dev = (struct amebasmart_usbdev_s *)priv;
+	DEBUGASSERT(dev);
+
+	usbd_cdc_acm_deinit();
+}
+
+static int amebasmart_usbd_init(struct amebasmart_usbdev_s *priv, usbd_config_t *cfg)
+{
+	struct amebasmart_usbdev_s *dev = (struct amebasmart_usbdev_s *)priv;
+
+	int ret = -1;
+	ret = usbd_init(cfg);
+
+	return ret;
+}
+
+/****************************************************************************
+ * Name: amebasmart_up_usbuninitialize
+ * Description: Initialize the USB driver
+ * Input Parameters: None
+ * Returned Value: None
+ ****************************************************************************/
+int amebasmart_up_usbinitialize(struct amebasmart_usbdev_s *priv)
+{
+	struct amebasmart_usbdev_s *dev = (struct amebasmart_usbdev_s *)priv;
+	DEBUGASSERT(dev);
+	
+	int ret = -1;
+
+	ret = amebasmart_usbd_init(dev, &amebasmart_cdc_acm_cfg);
+	if (ret != 0) {
+		dbg("usbd init fail\n");
+		return ret;
+	}
+
+	ret = amebasmart_usbd_cdc_acm_init(dev, &amebasmart_cdc_acm_cb);
+	if (ret != 0) {
+		dbg("usbd acm init fail\n");
+		return ret;
+	}
+	sem_init(&g_usbdev.txsem, 0, 0);
+
+	return ret;
+}
+
+/*It takes sometime for USB initialization to complete, so during boot up stage, it /dev/console will be
+ register with serial first to prevent opening an empty path, then once usb is ready, re-register it to usb and
+ open the fd*/
+static int register_usb(void)
+{
+	while(!cdc_acm_ready_flag){
+		usleep(1000);
+	}
+	USB_DEV.isconsole = true;
+	uart_register("/dev/ttyACM0", &USB_DEV);
+	unregister_driver("/dev/console");
```

Comment:
Let's add some comment on why we are unregistering this, eg; due to serial sharing same fd........